File: solutions/javascript/hamming/1/hamming.js
//
// This is only a SKELETON file for the 'Hamming' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const compute = (a, b) => {
  if (a.length !== b.length) {
    throw new Error("strands must be of equal length")
  }
  let distance = 0
  for (let i = 0; i < a.length; i++) {
    if (a[i] !== b[i]) {
      distance++
    }
  }
  return distance
};

File: solutions/c/hamming/1/hamming.c
#include "hamming.h"
#include <string.h>

int compute(const char *lhs, const char *rhs) {
    int n = strlen(lhs);
    int m = strlen(rhs);
    if (n != m)
        return -1;
    int distance = 0;
    for (int i = 0; i < n; i++)
        if (lhs[i] != rhs[i])
            distance++;
    return distance;
}